plugins/ssl/ssl-certificate-expiry

```diff
@@ -123,10 +126,33 @@ print_expire_days() {
     [ -n "$starttls" ] && s_client_args="-starttls $starttls"
 
     # shellcheck disable=SC2086
+
+    # The original code only extracted the server certificate,
+    # but the end date also depends on intermediate certs. Therefore
+    # we want to check intermediate certs as well.
+    #
+    # The following cryptic lines do:
+    # - invoke openssl and connect to a port
+    # - print certs, not only the server cert
+    # - extract each certificate as a single line
+    # - pipe each cert to the parse_valid_days_from_certificate
+    #   function, which basically is 'openssl x509 -enddate'
+    # - get a list of the parse_valid_days_from_certificate
+    #   results and sort them
     echo "" | openssl s_client \
-            -servername "$host" -connect "${host}:${port}" \
-            $s_client_args 2>/dev/null \
-        | parse_valid_days_from_certificate
+	-servername "$host" -connect "${host}:${port}" \
+	-showcerts \
+	$s_client_args 2>/dev/null | \
+	awk '{
+  	  if ($0 == "-----BEGIN CERTIFICATE-----") cert=""
+  	  else if ($0 == "-----END CERTIFICATE-----") print cert
+  	  else cert=cert$0
+	  }' | ( \
```

Comment:
I think, you can omit the sub-shell (via `( ... )`) here, since the `while ... done` construct is treated as a single command.